src/pool.rs

```diff
@@ -0,0 +1,27 @@
+use std::{sync, thread};
+
+pub(crate) type Task = dyn FnOnce() + Send;
+pub(crate) type BoxedTask = Box<Task>;
```

Comment:
I don't think these type aliases pull their weight. With my suggestion from a previous comment, they are only used in one spot. So please remove the aliases and just write out the type in the `scoped_run_tasks` function.